client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx

```diff
@@ -2,15 +2,27 @@ import React from 'react';
 import { format } from 'date-fns';
 import { useSelector } from 'react-redux';
 import { Typography } from '@material-ui/core';
+import { useFormContext } from 'react-hook-form'
 
 import ContactType from 'models/ContactType';
 import StoreStateType from 'redux/storeStateType';
 import InteractedContact from 'models/InteractedContact';
+import InvalidFormIcon from 'commons/Icons/InvalidFormIcon';
 import FamilyContactIcon from 'commons/Icons/FamilyContactIcon';
 import useInvolvedContact from 'Utils/vendor/useInvolvedContact';
 
+
 const ContactDetails = (props: Props) => {
-    const { interactedContact } = props;
+    const { errors } = useFormContext();
+    const { index , interactedContact } = props;
+
+    const formErrors = errors.form ? errors.form[index] : {};
+    let formHasErrors = false;
+    for(let key in formErrors) {
+        if(formErrors[key] !== undefined) {
+            formHasErrors = true;
```

Comment:
Instead of using let and changing it consider using state

Reply:
I don't want the function to trigger a re-render of the page, it's dependent on errors that depends on form and every change in the form re-triggers this function anyway

Reply:
tbh, maybe no even state but rather just a const constructed with .some()
  similarly to https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/998#discussion_r548542322

Reply:
fixed [here](7f40897)